backend/src/main/scala/petclinic/services/VisitServiceLive.scala

```diff
@@ -22,24 +34,47 @@ final case class VisitServiceLive(dataSource: DataSource, vetService: VetService
       _     <- run(query[Visit].insertValue(lift(visit))).provideEnvironment(ZEnvironment(dataSource))
     } yield visit
 
+  /** delete uses the filter method to find a Visit in the database with an ID
+    * that matches the one provided and deletes it. Unit is returned to indicate
+    * that we are running this method for its side effects, a deleted Visit
+    * gives us no information. This will either fail or succeed.
+    */
   override def delete(id: VisitId): Task[Unit] =
     run(query[Visit].filter(_.id == lift(id)).delete)
       .provideEnvironment(ZEnvironment(dataSource))
       .unit
 
+  /** get uses the filter method to find a Visit in the database with an ID tbat
+    * matches the one provided and returns it.
+    */
   override def get(id: VisitId): Task[Option[Visit]] =
     run(query[Visit].filter(_.id == lift(id)))
       .provideEnvironment(ZEnvironment(dataSource))
       .map(_.headOption)
 
+  /** getForPet uses the filter method to find all Visits in the database with a
+    * PetId that matches the one provided, returning all Visits associated with
+    * a given Pet.
+    */
   override def getForPet(petId: PetId): Task[List[Visit]] =
     run(query[Visit].filter(_.petId == lift(petId)).sortBy(_.date))
       .provideEnvironment(ZEnvironment(dataSource))
 
+  /** getAll uses the query method to find all entries in the database of type
+    * Visit and returns them.
+    */
   override def getAll: Task[List[Visit]] =
     run(query[Visit])
       .provideEnvironment(ZEnvironment(dataSource))
 
+  /** update uses the filter method to find a Visit in the database with an ID
+    * that matches the one provided and updates it with the provided optional
+    * values. Because a user may not provide all optional values, setOpt is used
+    * to preserve the existing value in the case one is not provided to replace
+    * it. Unit is returned to indicate side-effecting code. Note that this
+    * dynamicQuery is not generated at compile time.
+    * https://getquill.io/#writing-queries-dynamic-queries
+    */
```

Comment:
Try this format.

First, a high-level method description, and then a new paragraph where you go into implementation details:

```scala
/** Updates a `Visit` in the database with the given properties
  *
  * Uses the filter method to find a Visit in the database with an ID...
  */
```